checkov/policies_3d/checks_parser.py

```diff
@@ -18,3 +37,121 @@ def parse_raw_check(self, raw_check: Dict[str, Dict[str, Any]], **kwargs: Any) -
+        check = Base3dPolicyCheck()
+        self._fill_check_metadata(check)
+        check.predicaments = []
+
+        cve_predicaments = list(filter(None, [self._create_cve_predicament(cve_report) for cve_report in cves_reports]))
+        iac_predicaments = list(filter(None, [self._create_iac_predicament(iac_record) for iac_record in iac_records]))
+        secrets_predicaments = list(filter(None, [self._create_secrets_predicament(secrets_record) for secrets_record in secrets_records]))
+
+        # Generating all predicaments combinations while filtering empty lists
+        all_combinations = list(itertools.product(*filter(bool, [cve_predicaments, iac_predicaments, secrets_predicaments])))
+
+        for combination in all_combinations:
+            check.predicaments.append(
+                Predicament(
+                    logical_op='and',
+                    predicaments=[predicament for predicament in combination]
+                )
+            )
+
+        return check
+
+
+    def _create_cve_predicament(self, cve_report: ReportCVE) -> Predicament | None:
+        cve_definition = [definition["cves"] for definition in self.check_definition if "cves" in definition.keys()][0]
+        if not cve_definition:
+            return None
+
+        if not any(op in cve_definition.keys() for op in SUPPORTED_LOGICAL_OPERATORS):
+            return None
+
+        top_level_predicament = Predicament(logical_op='')
+        top_level_logical_op = ''
+        for key, value in cve_definition.items():
+            if key in SUPPORTED_LOGICAL_OPERATORS:
+                top_level_logical_op = key
+                top_level_predicament.logical_op = key
+            elif key == 'risk_factors':
+                value = [value] if isinstance(value, str) else value
+                top_level_predicament.predicates.append(RiskFactorCVEContains(value, cve_report))
+
+        nested_definition = cve_definition[top_level_logical_op]
+        nested_logical_op = None
+        for key, value in nested_definition[0].items():
+            if key in SUPPORTED_LOGICAL_OPERATORS:
+                nested_logical_op = key
+            elif key == 'risk_factors':
+                value = [value] if isinstance(value, str) else value
+                top_level_predicament.predicates.append(RiskFactorCVEContains(value, cve_report))
+
+        nested_predicament = None
+        if nested_logical_op:
+            nested_definition = nested_definition[0][nested_logical_op]
+            if not isinstance(nested_definition, list):
+                return None
+
+            nested_predicament = Predicament(logical_op=nested_logical_op)
+            for definition in nested_definition:
+                for key, value in definition.items():
+                    if key == 'risk_factors':
+                        value = [value] if isinstance(value, str) else value
+                        nested_predicament.predicates.append(RiskFactorCVEContains(value, cve_report))
```

Comment:
Maybe extract this to a separated function like `add_predicate(predicate_key, predicate_value, predicament)` or something similar. 2 benefits:
1. No duplicate code (this is very similar to lines 87-89)
2. In that way it can be a generic parsing so it can get any cve predicate (other than risk factor as well), and choose the right predicate by a map of key string to CVE predicate. That way adding new predicates would only require to add a new CVEPredicate inheritance, and this code would support it out of the box.

Reply:
extracted into `_create_predicate`